dbms/src/DataStreams/FormatFactory.cpp

```diff
@@ -92,6 +98,18 @@ BlockInputStreamPtr FormatFactory::getInput(const String & name, ReadBuffer & bu
     {
         return wrap_row_stream(std::make_shared<JSONEachRowRowInputStream>(buf, sample, settings.input_format_skip_unknown_fields));
     }
+#if USE_CAPNP
+    else if (name == "CapnProto")
+    {
+        std::vector<String> tokens;
+        auto schema_and_root = settings.format_schema.toString();
+        boost::split(tokens, schema_and_root, boost::is_any_of(":"));
+        if (tokens.size() != 2)
+            throw Exception("Format CapnProto requires 'format_schema' setting to have schema_file:root_object format, e.g. 'schema.capnp:Message'");
```

Comment:
Does the syntax `path:root` is somehow usual or recommended?
What if the path to file contains `:`?

Is it possible to pass any path? Then it should be checked to be somewhere inside clickhouse directory for security reasons (look at StorageFile for example).

Reply:
That's a good question, the schema file and root object are typically two separate arguments.
I didn't want to create another setting for it, but perhaps it'd be best.
Good point about enforcing local storage path, I'll add it.